### PR TITLE
backport PR #2631 to take into account additionalProperties set to false

### PR DESCRIPTION
**Pull request by Lausselloic**
backport https://github.com/rjsf-team/react-jsonschema-form/pull/2631 in v5

### CODE REVIEW

**Comment by heath-freenome** [comment]
@Lausselloic Sorry about the conflicts, but I needed to reformat the code to match the coding styles... Can you rebase and resolve the conflicts?

**Comment by Lausselloic** [comment]
@heath-freenome  I've update my PR to take the other fix. I was unable to build the entire packages locally so I wasn't able to run the test proposed here : https://github.com/rjsf-team/react-jsonschema-form/pull/2853/files#diff-b7cc040be2092e8b4493e2f99a74e1b5b43189cdd143f96ce81752c357462aa0L3212

@lpillonel does this fix sounds good to you too?

**Comment by lpillonel** [comment]
It looks ok for me yep

**Comment by heath-freenome** [comment]
@Lausselloic it looks like you'll need to run the `npm run cs-format` command on the PR to fix formatting issues. I would recommend fetching the latest `rjsf-v5` branch and rebase your code on top of it first.

**Comment by heath-freenome** [comment]
@Lausselloic You'll need to resolve conflicts now. Also do run the `npm run cs-format` tool as well

**Comment by heath-freenome** [comment]
@Lausselloic Just checking in to see whether you'll be able to update this PR before we merged the branch into master sometime next week

File: packages/utils/src/schema/toPathSchema.ts [hunk 1]
```diff
@@ -37,8 +37,9 @@ export default function toPathSchema<T = any>(
     $name: name.replace(/^\./, ""),
   } as PathSchema;
 
-  if (ADDITIONAL_PROPERTIES_KEY in schema) {
-    set(pathSchema, "__rjsf_additionalProperties", true);
+  if (ADDITIONAL_PROPERTIES_KEY in schema && 
+    schema[ADDITIONAL_PROPERTIES_KEY] === true) {
+    set(pathSchema, '__rjsf_additionalProperties', true);
```
Comment (heath-freenome) [inline packages/utils/src/schema/toPathSchema.ts:42]: These want to be double quotes and will be updated automatically by the formatter

Comment (heath-freenome) [inline packages/utils/src/schema/toPathSchema.ts:42]:
```suggestion
    set(pathSchema, "__rjsf_additionalProperties", true);
```

File: packages/utils/src/schema/toPathSchema.ts [hunk 2]
```diff
@@ -39,7 +39,8 @@ export default function toPathSchema<T = any>(
     [NAME_KEY]: name.replace(/^\./, ""),
   } as PathSchema;
 
-  if (ADDITIONAL_PROPERTIES_KEY in schema) {
+  if (ADDITIONAL_PROPERTIES_KEY in schema && 
+    schema[ADDITIONAL_PROPERTIES_KEY] === true) {
```
Comment (heath-freenome) [inline packages/utils/src/schema/toPathSchema.ts:43]:
```suggestion
  if (ADDITIONAL_PROPERTIES_KEY in schema && schema[ADDITIONAL_PROPERTIES_KEY] === true) {
```

**Pull request merged**